orangecontrib/bio/genialis/genapi.py

```diff
@@ -0,0 +1,190 @@
+from genesis import Genesis
+from genesis import GenData
+import json
+from Orange.data import *
+from .. import dicty
+import io
+import gzip
+
+DEFAULT_EMAIL = 'anonymous@genialis.com'
+DEFAULT_PASSWD = 'anonymous'
+DEFAULT_URL = 'https://dictyexpress.research.bcm.edu'
+CallBack = dicty.CallBack
+
+
+class Genapi(object):
+
+    """
+
+    Python module that leverages Genesis PyAPI (Python API for accsess to DictyExpress database).
+    It supports connection to the server and data retrieval functionalities.
+    """
+
+    def __init__(self, email=DEFAULT_EMAIL, password=DEFAULT_PASSWD, url=DEFAULT_URL, buffer=None):
+
+        self._gen = Genesis(email, password, url)
+        self.buffer = buffer
+        self.email = email
+
+    def fetch_etc_objects(self, callback=None):
+        """
+        Function downloads all available :obj:`GenData` etc objects from DictyExpress database.
+        Objects are stored in buffer for further use.
+        If buffer is set to None function returns a list of :obj:`GenData` objects.
+        """
+
+        def optcb():
+            if callback:
+                callback()
+
+        cbc = CallBack(1, optcb, callbacks=70)
+
+        store_experiments = []
+
+        list_of_experiments = self._gen.api.data.get(case_ids__contains='5535115cfad58d5e03006217', status='done',
```

Comment:
This project ID will be different on different servers. Maybe you should move it to the dict where you define available servers.